packages/js/src/plugins/tokenModule/addMetadata.ts

```diff
@@ -0,0 +1,141 @@
+import { ConfirmOptions, PublicKey } from '@solana/web3.js';
+import {
+  Collection,
+  Uses,
+  createCreateMetadataAccountV2Instruction,
+} from '@metaplex-foundation/mpl-token-metadata';
+import { Metaplex } from '@/Metaplex';
+import {
+  useOperation,
+  Operation,
+  Signer,
+  OperationHandler,
+  Creator,
+  BigNumber,
+  toUniformVerifiedCreators,
+} from '@/types';
+import { findMetadataPda } from '../nftModule/pdas';
+import { DisposableScope, Option, TransactionBuilder } from '@/utils';
+import { SendAndConfirmTransactionResponse } from '../rpcModule';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'AddMetadataOperation' as const;
+export const addMetadataOperation = useOperation<AddMetadataOperation>(Key);
+export type AddMetadataOperation = Operation<
+  typeof Key,
+  AddMetadataInput,
+  AddMetadataOutput
+>;
+
+export interface AddMetadataInput {
+  // Accounts.
+  mint: PublicKey;
+  payer?: Signer; // Defaults to mx.identity().
+  updateAuthority?: Signer; // Defaults to mx.identity().
```

Comment:
Here the `updateAuthority` does not need to be a Signer.

```suggestion
  updateAuthority?: PublicKey; // Defaults to mx.identity().publicKey.
```